mobile/src/screens/BookScreen.tsx

```diff
@@ -114,9 +164,25 @@ export const BookScreen: React.FC<BookScreenProps> = ({ route, navigation }) =>
     learn: i18n.t('learn'),
   };
 
+  // whether the books is favorited or not (solid vs empty bookmark)
+  const favoriteIconView =  favorited ? (
+
+    <TouchableOpacity style={styles.bookmarkContainer} onPress={() => {unfavoriteBook(book.id);}}>
+      <Image style={styles.bookmarkButton} source={require('../../assets/images/bookmark-solid.png')} />
+      <PopUpModal text={i18n.t('saved')} setModalVisible={setModalVisible} modalVisible={modalVisible} goBack={false}/>
```

Comment:
I actually don't think we really need this modal since the icon color changes on tap, let's remove the modal entirely for now.